mne/io/brainvision/brainvision.py

```diff
@@ -152,31 +169,32 @@ def __init__(self, vhdr_fname, montage=None,
         annots = read_annotations(mrk_fname, info['sfreq'])
         self.set_annotations(annots)
 
-        # Use events_from_annotations to properly set the events
-        trig_shift_by_type = _check_trig_shift_by_type(trig_shift_by_type)
-        dropped_desc = []  # use to collect dropped descriptions
-        event_id = dict() if event_id is None else event_id
-        event_id = partial(_event_id_func, event_id=event_id,
-                           trig_shift_by_type=trig_shift_by_type,
-                           dropped_desc=dropped_desc)
-        events, _ = events_from_annotations(self, event_id)
-        if len(dropped_desc) > 0:
-            dropped = list(set(dropped_desc))
-            warn("{0} event(s) will be dropped, such as {1}. "
-                 "Consider using the event_id parameter to parse events "
-                 "that do not follow the BrainVision format. For more "
-                 "information, see the docstring of read_raw_brainvision."
-                 .format(len(dropped), dropped[:5]))
-        self._create_event_ch(events, n_samples)
+        if stim_channel:
+            # Use events_from_annotations to properly set the events
+            trig_shift_by_type = _check_trig_shift_by_type(trig_shift_by_type)
+            dropped_desc = []  # use to collect dropped descriptions
+            event_id = dict() if event_id is None else event_id
+            event_id = partial(_event_id_func, event_id=event_id,
+                               trig_shift_by_type=trig_shift_by_type,
+                               dropped_desc=dropped_desc)
+            events, _ = events_from_annotations(self, event_id)
+            if len(dropped_desc) > 0:
+                dropped = list(set(dropped_desc))
+                warn("{0} event(s) will be dropped, such as {1}. "
+                     "Consider using the event_id parameter to parse events "
+                     "that do not follow the BrainVision format. For more "
+                     "information, see the docstring of read_raw_brainvision."
+                     .format(len(dropped), dropped[:5]))
+            self._create_event_ch(events, n_samples)
```

Comment:
Might be a good time to refactor and share this with the similar code in the EEGLAB reader.

Reply:
These lines will be removed as soon as 0.17 is released so it's not worth it